### PR TITLE
add support for powershell

### PR DESCRIPTION
**Pull request by paulygreen**
I use powershell with conda and the activate.d and deactivate.d scripts generated on win32 don't work properly (they seem to work for cmd.exe shells only). This PR adds functionality to generate .ps1 scripts (in addition to the default bat files) if the --generate-ps1 flag is passed on the command line. Without --generate-ps1, the output and generated scripts are unchanged.

changes:
- add new argument 'generate-ps1' to generate activate.d and deactivate.d scripts for powershell 
- updated gen_scripts.py to support powershell (ps1) scripts 
- added unit tests for powershell

### CODE REVIEW

File: conda_devenv/devenv.py [hunk 1]
```diff
@@ -454,6 +454,8 @@ def write_activate_deactivate_scripts(
 
     if sys.platform == "win32":
         files = [("devenv-vars.bat", "cmd")]
+        if args.generate_ps1:
+            files += [("devenv-vars.ps1", "ps1")]
     else:
         # Linux and Mac should create a .sh
         files = [("devenv-vars.sh", "bash"), ("devenv-vars.fish", "fish")]
```
Comment (prusse-martin) [inline conda_devenv/devenv.py:461]: Was it a problem to always generate them?
In linux we always generate 2 sets of scripts, one for bash and one for fish.

**Comment by paulygreen** [comment]
No I don’t think it’s a problem. I just didn’t want to change anything
“unnecessarily”. I tested it on my machine and having both .bat and .ps1
scripts didn’t seem to interact with each other. If I used cmd.exe shell it
loaded the bat script and if I used powershell it would use the ps1
scripts.

Ironically, not changing anything “unnecessarily” required adding a cli
argument. I guess it would be more transparent to just always generate ps1
scripts on win32.

Paul

File: conda_devenv/devenv.py (hunk 1 above)
Comment (paulygreen) [reply conda_devenv/devenv.py]: I updated the PR to always generate ps1 scripts for win32

**Review by prusse-martin — APPROVED** [review]

**Comment by prusse-martin** [comment]
Thank you for the PR.
I forgot to ask to you update the change log =)
I already did update that and also started a release process in #151.

**Pull request merged**